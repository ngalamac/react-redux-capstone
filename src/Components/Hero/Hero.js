import React from 'react';
import cover from '../../Assets/Img/cover1.jpg';

const Hero = () => (

  <>
    <div className="hero__container">
      <img
        src={cover}
        alt="air pollution cover"
        className="hero__cover-image"
      />
      <p className="hero__cover-text">Air Pollution App</p>
    </div>
  </>

);

export default Hero;
