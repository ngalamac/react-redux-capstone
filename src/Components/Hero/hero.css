.hero__container {
  width: 100%;
  position: relative;
  background-color: var(--color-secondary-dark);
}

.hero__cover-image {
  width: 100%;
  height: 100%;
  filter: invert(0%) sepia(50%) saturate(200%) hue-rotate(181deg) brightness(100%) contrast(100%);
}

.hero__cover-text {
  position: absolute;
  bottom: 24px;
  left: 24px;
  color: var(--color-white);
  font-family: var(--font-titles);
  font-weight: 500;
  font-size: 2rem;
}

.hero__continent-title {
  background-color: var(--color-secondary-dark);
  color: var(--color-white);
  font-weight: 500;
  padding-left: 2rem;
  padding-bottom: 0.4rem;
}

@media only screen and (min-width: 768px) {
  .hero__container {
    width: 50%;
  }

  .hero__continent-title {
    background-color: var(--color-primary-ligth);
    color: var(--color-white);
    font-weight: 500;
    font-size: 2rem;
  }
}
