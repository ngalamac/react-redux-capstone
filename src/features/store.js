import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './countries/countrySlice';

const store = configureStore({
  reducer: {
    country: countryReducer,
  },
});

export default store;
