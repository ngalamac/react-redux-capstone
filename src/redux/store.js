import { configureStore } from '@reduxjs/toolkit';
import teamsReducer from './footballSlice';

const store = configureStore({
  reducer: {
    teams: teamsReducer,
  },
});

export default store;
