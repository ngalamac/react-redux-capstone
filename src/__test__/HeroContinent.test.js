import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Hero from '../Components/Hero/Hero';
import store from '../Redux/configureStore';

describe('The indivial page of the app countires', () => {
  it('renders properly', async () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Hero />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
