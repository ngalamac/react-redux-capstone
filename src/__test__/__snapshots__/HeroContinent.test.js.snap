// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The indivial page of the app countires renders properly 1`] = `
<div
  className="hero__container"
>
  <img
    alt="air pollution cover"
    className="hero__cover-image"
    src="cover1.jpg"
  />
  <p
    className="hero__cover-text"
  >
    Air Pollution App
  </p>
</div>
`;
